core/cog-directory-files-handler.c

```diff
@@ -186,6 +188,14 @@ cog_directory_files_handler_run (CogRequestHandler      *request_handler,
     while (path[0] == '/')
         ++path;
 
+    /* Discard non-empty leading path components. */
```

Comment:
This logic is not trivial. Not for my eyes at least. It could be nice if you could add here what is the expected behavior with some specific examples.

Reply:
Comment added to explain things a bit, I hope it makes the code easier to follow.